src/locale/locale.tsx

```diff
@@ -1,11 +1,13 @@
-import React, {createContext, useMemo, useContext} from 'react';
-import {I18n, I18nManager} from '@shopify/react-i18n';
+import React, {createContext, useMemo, useContext, useRef, useEffect} from 'react';
+import {I18n as ReactI18n, I18nManager} from '@shopify/react-i18n';
 import {createStorageService, useStorage} from 'services/StorageService';
 import {captureException} from 'shared/log';
 
 import LOCALES from './translations';
 import FALLBACK_LOCALE from './translations/en.json';
 
+export type I18n = Pick<ReactI18n, 'locale' | 'translate'>;
```

Comment:
Because we create a wrapper for I18n, we can scope it down to just what we need. So, we pick only `locale` and `translate`.